src/composites.rs

```diff
@@ -103,64 +99,70 @@ where
         clippy::new_without_default_derive
     )
 )]
-impl<U: GetPath, V: NotFound, W> CompositeNewService<U, V, W> {
-    /// create an empty `CompositeNewService`
+impl<C, U, V: NotFound<V>, W> CompositeMakeService<C, U, V, W> {
+    /// create an empty `CompositeMakeService`
     pub fn new() -> Self {
-        CompositeNewService(Vec::new())
+        CompositeMakeService(Vec::new())
     }
 }
 
-impl<U, V, W> NewService for CompositeNewService<U, V, W>
+impl<C, U, V, W> MakeService<C> for CompositeMakeService<C, U, V, W>
 where
-    U: GetPath,
-    V: NotFound + 'static,
-    W: 'static,
+    U: hyper::body::Payload,
+    V: NotFound<V> + 'static + hyper::body::Payload,
+    W: std::error::Error + Send + Sync + 'static,
 {
-    type Request = U;
-    type Response = V;
+    type ReqBody = U;
+    type ResBody = V;
     type Error = W;
-    type Instance = CompositeService<U, V, W>;
-
-    fn new_service(&self) -> Result<Self::Instance, io::Error> {
+    type Service = CompositeService<U, V, W>;
+    type MakeError = io::Error;
+    type Future = futures::future::FutureResult<Self::Service, io::Error>;
+
+    fn make_service(
+        &mut self,
```

Comment:
Because we are implementing `hyper::service::MakeService<C>` which defines this parameter as mutable - https://docs.rs/hyper/0.12.25/hyper/service/trait.MakeService.html#tymethod.make_service

Reply:
OK.